src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsF2Uap.java

```diff
@@ -0,0 +1,93 @@
+public class TestAdsF2Uap extends AdsF2TestTemplate {
+
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomArticleUapPage",
+      groups = {"AdsF2UapDesktop"}
+  )
+  public void adsFandomArticleUapDesktop(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    verifyUapBtf(btfId, AdsFandomContent.INCONTENT_BOXAD_F2, AdsFandomContent.BOTTOM_LEADERBOARD_F2, fandomPage);
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomArticleUapPage",
+      groups = {"AdsF2UapMobile"}
+  )
+  @RelatedIssue(issueID = "ADEN-4339")
+  public void adsFandomArticleUapMobile(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapBtf(btfId, AdsFandomContent.INCONTENT_BOXAD_F2, AdsFandomContent.BOTTOM_LEADERBOARD_F2, fandomPage);
+  }
+
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomTopicPage",
+      groups = {"AdsF2UapDesktop"}
+  )
+  public void adsF2UapOnTopicDesktop(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    verifyUapBtf(btfId, AdsFandomContent.FEED_BOXAD, AdsFandomContent.BOTTOM_LEADERBOARD_F2, fandomPage);
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      dataProviderClass = F2AdsDataProvider.class,
+      dataProvider = "fandomTopicPage",
+      groups = {"AdsF2UapMobile"}
+  )
+  @RelatedIssue(issueID = "ADEN-4339")
+  public void adsF2UapOnTopicMobile(String pageType, String pageName, long atfId, long btfId) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    fandomPage.triggerOnScrollSlots();
+    verifyUapAtf(atfId, AdsFandomContent.TOP_BOXAD, fandomPage);
+    verifyUapBtf(btfId, AdsFandomContent.FEED_BOXAD, AdsFandomContent.BOTTOM_LEADERBOARD_F2, fandomPage);
+  }
+
+  private void verifyUapAtf(long atfId, String slotName, AdsFandomObject fandomPage) {
+    fandomPage.verifySlot(AdsFandomContent.TOP_LEADERBOARD);
+    Assertion.assertEquals(atfId, fandomPage.getLineItemId(AdsFandomContent.TOP_LEADERBOARD));
+
+    fandomPage.verifySlot(slotName);
+    Assertion.assertEquals(atfId, fandomPage.getLineItemId(AdsFandomContent.TOP_BOXAD));
+  }
+
+  private void verifyUapBtf(long btfId, String slotName, String bottomSlotName, AdsFandomObject fandomPage) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "verifyUapBtf" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)